### ISSUE TITLE
Different beaviour in BrandHeader/resize

### ISSUE THREAD

**Issue by tommy-codez** [telekom/scale#556]
Hi there,
it seems the Brand Header Main Menu acting differently depending on the implementation.

First implementation
```
<ScaleAppShell mainNavigation={[
      {
        name: "Custom Item #1",
        id: "Custom Item #1",
        href: "#topic-one",
      },
    {
        name: "Custom Item #2",
        id: "Custom Item #2",
        href: "#topic-two",
      },
      {
        name: "Custom Item #3",
        id: "Custom Item #3",
        href: "#topic-three",
      },
      {
        name: "Custom Item #4",
        id: "Custom Item #4",
        href: "#topic-four"
      },
  
  
    ]}>
    <p style={{ height: '500vh' }}>some large content</p>
      </ScaleAppShell>
```
![grafik](https://user-images.githubusercontent.com/66920189/131487657-3d7722c0-6848-4568-a8f0-f798f63d6eec.png)

During a resize of the window the menu completely disappears and a mini menu appears.

Second Implementation

```

  <ScaleAppShell>
	  <ScaleAppHeader slot="header" id="header">
		<div slot="menu-main">
		<ScaleNavMain name="Custom Item #1"></ScaleNavMain>
			<ScaleNavMain name="Custom Item #2" id="nav-main-with-mega-menu">
			</ScaleNavMain>
			<ScaleNavMain name="Custom Item #3"></ScaleNavMain>
			<ScaleNavMain name="Custom Item #4"></ScaleNavMain>
		</div>
	  </ScaleAppHeader>
	  <p style={{ height: '500vh' }}>some large content</p>
	</ScaleAppShell>

```

![grafik](https://user-images.githubusercontent.com/66920189/131488422-cfabf7e6-90bb-41fd-adbe-d6953fb4ee0c.png)

here there is no main menu and no mini menu during a resize of the window

**Comment by nowseemee** [comment]
Hi @tommy-codez, thanks for your question.

You can either pass the mainNavigation config object to ScaleAppHeader or implement your own mobile menu:

```
<body>
  <scale-app-shell>
    <scale-app-header slot="header" id="header">
      <div slot="menu-mobile"><pre>menu-mobile</pre></div>
    </scale-app-header>
    <div style="height: 800px; padding: 48px 0;">
      Scroll to test header behavior
    </div>
    <div style="height: 800px; padding: 48px 0;">
      Scroll more to test header behavior
    </div>
  </scale-app-shell>
  <script>
    const header = document.getElementById('header');
    document.body.addEventListener('keydown', function (event) {
      if (['Escape', 'Esc'].includes(event.key)) {
        event.preventDefault();
        header.dispatchEvent(new Event('closeMenu'));
      }
    });
  </script>
</body>
```

Unfortunately, there is no way to automatically render a mobile menu when only the custom `menu-main` slot is provided so you should go with one of the above options. 

Feel free to reopen this issue in case you have more questions.

**Comment by tommy-codez** [comment]
Hm I am not sure if I understand. So the mobile menu is only available with a config object?

So how can I customize the header if I want to add for example a dropdown menu and also want the mobile menu ?

**Comment by nowseemee** [comment]
It is possible to use the `menu-main` slot along with the config object so you will have your dropdown menu in the desktop version and a standard mobile menu rendered based on the config you pass when you switch to a small screen. In case you have special requirements for the mobile menu, you can use the `menu-mobile` and implement the custom requirements. I hope this helps.

**Comment by tommy-codez** [comment]
Thanks for your help - but as soon I want to customize the header it resets what I have in the config object - perhaps I am doing something wrong ;)


  <ScaleAppShell
      mainNavigation={mainNavigation}
      iconNavigation={iconNavigation}
    >
      <ScaleAppHeader slot="header" id="header">
        <div slot="menu-addon">
          <ScaleNavSegment is-active="true" href="#custom-addon-1">
            Custom Addon #1
          </ScaleNavSegment>
          <ScaleNavSegment href="#custom-addon-2">
            Custom Addon #2
          </ScaleNavSegment>
        </div>
        <ScaleDropdown label="Small" size="small">
          <option value="1">item 1</option>
          <option value="2" selected>
            item 2
          </option>
          <option value="3">item 3</option>
        </ScaleDropdown>
      </ScaleAppHeader>

**Comment by nowseemee** [comment]
Yes, you are right - as soon as `menu-main` is provided, the config object is not considered for the mobile menu. I'll remove this limitation and publish a new version soon. Thanks for trying!

**Comment by nowseemee** [comment]
It's released: `3.0.0-beta.20` - many thanks for bringing this up!
